module/VuFindConsole/src/VuFindConsole/Command/Util/UpdateResourceMetadataCommand.php

```diff
@@ -0,0 +1,187 @@
+     *
+     * @param ResourceServiceInterface $resourceService    Resource service
+     * @param Loader                   $recordLoader       Record loader
+     * @param ResourcePopulator        $resourcePopulator  Resource pop
+     * @param PersistenceManager       $persistenceManager Persistence manager
+     * @param ?string                  $name               The name of the command; passing null means
+     * it must be set in configure()
+     */
+    public function __construct(
+        protected ResourceServiceInterface $resourceService,
+        protected Loader $recordLoader,
+        protected ResourcePopulator $resourcePopulator,
+        protected PersistenceManager $persistenceManager,
+        ?string $name = null,
+    ) {
+        parent::__construct($name);
+    }
+
+    /**
+     * Configure the command.
+     *
+     * @return void
+     */
+    protected function configure()
+    {
+        $this
+            ->setHelp('Updates the metadata fields of the resource table.')
+            ->addOption(
+                'batch',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Number of records to process in a single batch',
+                100
+            )
+            ->addOption(
+                'min-age',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Minimum age of a record (in days) before it is refreshed even if metadata is not missing.'
+                . ' By default records with missing metadata are updated, but using this option allows updates to all'
+                . ' records periodically.'
+            )->addOption(
+                'backend',
+                null,
+                InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+                'Record backend (source) to check. By default resources for all backends are checked.'
+            );
+    }
+
+    /**
+     * Run the command.
+     *
+     * @param InputInterface  $input  Input object
+     * @param OutputInterface $output Output object
+     *
+     * @return int 0 for success
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $batch = $input->getOption('batch');
+        $minAge = $input->getOption('min-age');
+        $backend = $input->getOption('backend');
+
+        $this->recordLoader->setCacheContext(Cache::CONTEXT_FAVORITE);
+
+        $updated = 0;
+        $redirected = 0;
+        $missing = 0;
+        $lastId = null;
+        $output->writeln('<info>Updating resource metadata</info>');
+        while ($resources = $this->resourceService->findMetadataToUpdate($lastId, $batch, $minAge, $backend)) {
+            $recordIds = [];
+            array_walk(
+                $resources,
+                function ($resource) use (&$recordIds) {
+                    $recordIds[] = ['id' => $resource->getRecordId(), 'source' => $resource->getSource()];
+                }
+            );
```

Comment:
Sorry if I'm missing some subtlety that makes this a bad idea, but why not simplify to this?


```suggestion
            $recordIds = array_map(
                fn ($resource) => ['id' => $resource->getRecordId(), 'source' => $resource->getSource()],
                $resources
            );
```